nomos-services/mempool/src/lib.rs

```diff
@@ -317,4 +334,5 @@ where
 pub struct Settings<B, N> {
     pub backend: B,
     pub network: N,
+    pub registry: Option<NomosRegistry>,
```

Comment:
Missing gate?

Reply:
One approach would be to feature gate everything completely, another one would only target the functionality.
In this case I was targeting only unnecessary functionality, reasoning being that Settings can have everything in them, but the service would act on those only if the feature is enabled.

If we want complete feature gate, eliminate dependencies etc then bit more changes are needed.

Reply:
Oh, ok. I get the approach now. Uhm, I do not see the harm on having it in the settings, although it feels a bit of a leftover. We can go with this for now. We can harden it later if needed (probably good to have an issue for tracking this).

Reply:
Created here: [nomos-node#523](https://github.com/logos-co/nomos-node/issues/523#issue-1987385478)